### ISSUE TITLE
dataclass_wizard.errors.UnknownJSONKey: A JSON key is missing from the dataclass schema for class `UploadResponse`.   unknown key: 'archived'

### ISSUE THREAD

**Issue by KaneDM** [rnag/wystia#59]
* Python version: 3.11
* Operating System: Windows 10

### Description

Attempted to upload a video to Wistia using a script I wrote that has worked in the past. Wystia crashed because the "UploadResponse" class didn't have a way to handle the "archived" key in the JSON returned from the Wistia API.

Fixed it by adding "archived" keys to the "UploadResponse" and "Video" classes in model.py.
Line 244 (Video class):
`archived: bool | None = None`
Line 744 (UploadResponse class):
`archived: bool`

### What I Did

Initial crash traceback:
```
Traceback (most recent call last):
  File "D:\Python\tvid.py", line 65, in <module>
    vidupload = WistiaUploadApi.upload_file(vidlatest, project, vidname)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\wystia\api_upload.py", line 74, in upload_file
    return UploadResponse.from_dict(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\dataclass_wizard\loaders.py", line 536, in fromdict
    return load(d)
           ^^^^^^^
  File "C:\Python311\Lib\site-packages\dataclass_wizard\loaders.py", line 611, in cls_fromdict
    field_name = lookup_field_for_json_key(o, json_key)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\dataclass_wizard\loaders.py", line 705, in lookup_field_for_json_key
    raise e from None
dataclass_wizard.errors.UnknownJSONKey: A JSON key is missing from the dataclass schema for class `UploadResponse`.
  unknown key: 'archived'
  dataclass fields: ['hashed_id', 'id', 'name', 'type', 'description', 'account_id', 'created', 'updated', 'progress', 'thumbnail', 'duration', 'status']
  input JSON object: {"id": 102354918, "name": "RT Wire Lengths", "type": "Video", "archived": false, "created": "2023-03-17T15:35:37+00:00", "updated": "2023-03-17T15:35:37+00:00", "duration": 1199.032022, "hashed_id": "malb60y0o2", "description": "", "progress": 0.125, "status": "queued", "thumbnail": {"url": "https://embed-ssl.wistia.com/deliveries/a3b1fbd6df17258c467c27207d45a1c5.jpg?image_crop_resized=200x120&video_still_time=599", "width": 200, "height": 120}, "account_id": 392724}
```

**Comment by rnag** [comment]
Hi @KaneDM ! Thanks for opening this issue. In hindsight, I do agree that enabling `raise_on_unknown_json_key` by default might not have been the best way to go.

I will see if I am able to publish a new release (with the recommended changes) shortly. In the meantime, I recommend checking out the [Contributing](https://wystia.readthedocs.io/en/latest/contributing.html#get-started) docs, as that also has info on how to contribute to the project and create and submit a PR for review, in case this is of interest.

Else, I'll likely just open a PR shortly, after adding and testing out the recommended changes. By the way, I assume you are OK with being credited and/or mentioned in the changelog for the release notes?

**Comment by rnag** [comment]
BTW, I'll also need to look into adding (and testing) the same new fields to a similar project I've published in Rust, at [`rust-wistia`](https://github.com/rnag/rust-wistia). I'll do this when I have time, either this weekend or the next 🤞 .

**Comment by KaneDM** [comment]
> Hi @KaneDM ! Thanks for opening this issue. In hindsight, I do agree that enabling `raise_on_unknown_json_key` by default might not have been the best way to go.

You're welcome! Love your library, it's worked without issue until now. And I agree that having that raise seems unusual for this case, but there might be a use case that's not obvious to me.

> I will see if I am able to publish a new release (with the recommended changes) shortly. In the meantime, I recommend checking out the [Contributing](https://wystia.readthedocs.io/en/latest/contributing.html#get-started) docs, as that also has info on how to contribute to the project and create and submit a PR for review, in case this is of interest.

Yeah, I know next to nothing about GitHub, this seemed to be the easiest way to bring it to your attention. lol

> Else, I'll likely just open a PR shortly, after adding and testing out the recommended changes. By the way, I assume you are OK with being credited and/or mentioned in the changelog for the release notes?

I am completely fine with all of that.

**Comment by rnag** [comment]
@KaneDM success 🎉 ! This issue should be addressed as part of the `v1.2.0` release. Please update the pinned dependency of the `wystia` library.

Here is the full diff:
https://github.com/rnag/wystia/compare/4944466...v1.2.0
